compiler/src/compile.rs

```diff
@@ -1481,6 +1481,7 @@ impl Compiler {
                 self.set_label(end_label);
             }
         }
+        self.optimize_instruction();
```

Comment:
Instead of calling this function here, call it in the `emit` function. The emit function will be called whenever there is a new function emitted to the output.